### PR TITLE
Updates for pysam 0.8.5.

### PR DESCRIPTION
**Pull request by jeromekelleher**
Uses the local download facility for BAM indexes.

-1: This shouldn't be merged until pysam 0.8.5 is actually released.

### CODE REVIEW

File: scripts/download_data.py [hunk 1]
```diff
@@ -500,11 +512,13 @@ def parseArgs():
...
-    downloader.downloadBams()
-    downloader.downloadFastas()
-    downloader.cleanup()
+    try:
+        downloader = downloaderClass(args)
+        downloader.downloadVcfs()
+        downloader.downloadBams()
+        downloader.downloadFastas()
+    finally:
+        downloader.cleanup()
```
Comment (jeromekelleher) [inline scripts/download_data.py]: Hmm, realised that this means that we always delete the checkpoint files too. Hopefully we're not going to need them any more now that htslib's networking code has been rewritten. Thoughts @dcolligan?

Comment (dcolligan) [inline scripts/download_data.py]: What's the problem with leaving the checkpoint functionality in?  The network connection could always drop in the middle of downloading.  And, although I hope for network improvements in this version of htslib, I doubt it has been purged of all bugs.

Comment (jeromekelleher) [inline scripts/download_data.py]: No problem, I just forgot about it. I'll fix it before we merge.

**Comment by jeromekelleher** [comment]
@dcolligan, pysam 0.9.0 has been released, so we should update. I think I've fixed the issues with breaking the checkpointing code too.

**Comment by jeromekelleher** [comment]
OK, hopefully this fixes all the bugs. Changes required are mostly straightforward encoding for bytes objects. @david4096, I ended up doing a bit of a refactor while I was changing the variant annotations code. Can you have a look at see if it makes sense please?

**Comment by dcolligan** [comment]
+1

**Comment by jeromekelleher** [comment]
Looks like this is ready to go. @david4096, can you review and merge if you're happy please?

**Comment by david4096** [comment]
-1 Looks like it breaks for the compliance data.
```
ERROR in app [/Users/david/server-1/server-1/lib/python2.7/site-packages/flask/app.py:1423]:
Exception on /variantannotations/search [POST]
--------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/david/server-1/server-1/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/david/server-1/server-1/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/david/server-1/ga4gh/frontend.py", line 519, in searchVariantAnnotations
    flask.request, app.backend.runSearchVariantAnnotations)
  File "/Users/david/server-1/ga4gh/frontend.py", line 407, in handleFlaskPostRequest
    return handleHttpPost(flaskRequest, endpoint)
  File "/Users/david/server-1/ga4gh/frontend.py", line 284, in handleHttpPost
    responseStr = endpoint(request.get_data())
  File "/Users/david/server-1/ga4gh/backend.py", line 814, in runSearchVariantAnnotations
    self.variantAnnotationsGenerator)
  File "/Users/david/server-1/ga4gh/backend.py", line 616, in runSearchRequest
    for obj, nextPageToken in objectGenerator(request):
  File "/Users/david/server-1/ga4gh/backend.py", line 552, in variantAnnotationsGenerator
    request, variantAnnotationSet)
  File "/Users/david/server-1/ga4gh/backend.py", line 205, in __init__
    request, parentContainer)
  File "/Users/david/server-1/ga4gh/backend.py", line 69, in __init__
    self._initialiseIteration()
  File "/Users/david/server-1/ga4gh/backend.py", line 82, in _initialiseIteration
    self._currentObject = next(self._searchIterator, None)
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 848, in getVariantAnnotations
    record, transcriptConverter)
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 1039, in convertVariantAnnotation
    transcriptEffects.append(transcriptConverter(ann, altshgvsG))
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 984, in convertTranscriptEffectSnpEff
    cdsPos, protPos, distance, errsWarns) = annStr.split('|')
```

**Comment by jeromekelleher** [comment]
Thanks @david4096, looks like we're missing some data from our data driven tests. Can you create an issue to describe the data we need to add please @david4096?

**Comment by david4096** [comment]
There are enough data driven tests in there but they aren't running correctly apparently. On my local machine it fails nosetests.

```
======================================================================
ERROR: tests.datadriven.test_variant_annotations.VariantAnnotationSetTest.testVariantsValid:WASH7P_annotation
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/david/server-1/tests/datadriven/__init__.py", line 29, in testFunction
    method()
  File "/Users/david/server-1/tests/datadriven/test_variant_annotations.py", line 170, in testVariantsValid
    for gaVariantAnnotation in iterator:
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 848, in getVariantAnnotations
    record, transcriptConverter)
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 1039, in convertVariantAnnotation
    transcriptEffects.append(transcriptConverter(ann, altshgvsG))
  File "/Users/david/server-1/ga4gh/datamodel/variants.py", line 984, in convertTranscriptEffectSnpEff
    cdsPos, protPos, distance, errsWarns) = annStr.split('|')
ValueError: need more than 1 value to unpack
```

I'm looking now to see what's going on in terms of my environment. pysam builds htslib/samtools itself, doesn't it?

**Comment by david4096** [comment]
The data are there for all the data driven tests and I am struggling to recreate the issue I mentioned above :8ball: . I am presuming pysam in my virtual environment wasn't updated or I had some old *.pyc lying around. You can rebase/merge at your own delight/peril :).

**Comment by jeromekelleher** [comment]
Excellent, thanks for checking @david4096. Merging.

**Pull request merged**